### PR TITLE
Add Sentry to the PPR UI for handling errors

### PR DESCRIPTION
**Pull request by jguertin**

### CODE REVIEW

File: ppr-ui/public/config/stub-configuration.json [hunk 1]
```diff
@@ -3,5 +3,7 @@
   "AUTH_URL": "http://localhost:8080/auth/",
   "AUTH_API_URL": "https://auth-api-dev.pathfinder.gov.bc.ca/api/v1/",
   "PAY_API_URL": "",
-  "ADDRESS_COMPLETE_KEY": "RH99-RN99-GB72-FW86"
+  "ADDRESS_COMPLETE_KEY": "RH99-RN99-GB72-FW86",
+  "SENTRY_DSN": "",
+  "SENTRY_ENVIRONMENT": "local-developer"
```
Comment (jguertin) [inline ppr-ui/public/config/stub-configuration.json]: I updated the config map in dev.  We can have a private chat about setting these values.

File: ppr-ui/src/main.ts [hunk 2]
```diff
@@ -30,13 +32,19 @@ Vue.config.errorHandler = (err, vm, info) => {
   console.error('Vue main error handler', err, vm, info)
 };
 
-window.onerror = function (message, source, lineno, colno, error) {
-  console.error('window.onerror', message, source, lineno, colno, error)
-}
```
Comment (jguertin) [inline ppr-ui/src/main.ts:41]: With the other error handling in place, this seemed redundant.

File: ppr-ui/src/utils/config-helper.ts [hunk 3]
```diff
@@ -14,31 +32,33 @@ export default {
...
-        console.log('Set Pay API URL to: ' + payApiUrl)
-
-        // TODO use this when ppr needs canada post address lookup.  Also need to fix the type checking error
-        // Element implicitly has an 'any' type because index expression is not of type 'number'.
-        // on the window['key'] statement
-        // const addressCompleteKey = response.data['ADDRESS_COMPLETE_KEY']
-        // window['addressCompleteKey'] = addressCompleteKey
-        // console.log('Set Address Complete Key')
-      })
+    const response = await axios.get(url, {headers})
```
Comment (jguertin) [inline ppr-ui/src/utils/config-helper.ts:41]: While I was in here, I adapted for `async/await`.  Personally, I strongly prefer this pattern over promises and chaining but am okay if you would like me to revert.  It's quick and easy,

File: ppr-ui/src/utils/config-helper.ts [hunk 4]
```diff
@@ -1,10 +1,28 @@
...
+
+  constructor(apiUrl: string, authUrl: string, authApiUrl: string, payApiUrl: string, sentryDSN: string, sentryEnvironment: string) {
+    this.apiUrl = apiUrl
+    this.authUrl = authUrl
+    this.authApiUrl = authApiUrl
+    this.payApiUrl = payApiUrl
+    this.sentryDSN = sentryDSN
+    this.sentryEnvironment = sentryEnvironment
+  }
+}
```
Comment (jguertin) [inline ppr-ui/src/utils/config-helper.ts]: Added a class to hold the config settings in an immutable fashion.  Generally speaking, I tend to make classes immutable unless there is a good reason not to.

**Review by bryan-gilbert — APPROVED** [review]

File: ppr-ui/public/config/sample-dev-configuration.json [hunk 5]
```diff
@@ -2,5 +2,7 @@
   "API_URL": "https://ppr-api-dev.pathfinder.gov.bc.ca/",
   "AUTH_URL": "https://dev.bcregistry.ca/ppr/auth/",
   "AUTH_API_URL": "https://auth-api-dev.pathfinder.gov.bc.ca/api/v1/",
-  "PAY_API_URL": "https://pay-api-dev.pathfinder.gov.bc.ca/api/v1/"
+  "PAY_API_URL": "https://pay-api-dev.pathfinder.gov.bc.ca/api/v1/",
+  "SENTRY_DSN": "",
+  "SENTRY_ENVIRONMENT": "dev"
```
Comment (jguertin) [inline ppr-ui/public/config/sample-dev-configuration.json:7]: Do we need this file?  I'm not clear about the purpose, as it doesn't seem to be used.

File: ppr-ui/src/utils/app-data.ts [hunk 6]
```diff
@@ -24,13 +24,21 @@ class FeatureFlags {
...
   private _pprApiUrl: string = sessionStorage.getItem('PPR_API_URL') || ''
 
+  readonly sentryDSN: string
+  readonly sentryEnvironment: string
+
+  constructor(data: object) {
+    this.sentryDSN = data['SENTRY_DSN']
+    this.sentryEnvironment = data['SENTRY_ENVIRONMENT'];
+  }
+
```
Comment (jguertin) [inline ppr-ui/src/utils/app-data.ts:41]: This is a discussion point over how we want to handle this kind of data.  I didn't want to do a sweeping change at this point, but do want to propose a pattern of immutability.  Also for discussion, when do we and don't we use session storage.

I'd like to see us wait until after we fetch the configuration.json before we initialize Config, and would like to see it remain immutable.  This impacts the static accessor below.

So the question is what's the pattern we want, and do I need to finish refactoring fully as part of this PR.

**Review by WalterMoar — APPROVED** [review]

**Pull request merged**